src/app/vendors/VendorItemComponent.tsx

```diff
@@ -46,19 +46,22 @@ export default function VendorItemComponent({
     item.item.collectibleState !== undefined &&
     !(item.item.collectibleState & DestinyCollectibleState.NotAcquired);
 
-  const unavailable =
-    !item.canPurchase ||
-    !item.canBeSold ||
-    (owned &&
-      item.item?.itemCategoryHashes.includes(ItemCategoryHashes.Bounties) &&
+  // Can't buy more copies of emblems or bounties other than repeatables.
+  const ownershipRule =
+    item.item?.itemCategoryHashes.includes(ItemCategoryHashes.Emblems) ||
+    (item.item?.itemCategoryHashes.includes(ItemCategoryHashes.Bounties) &&
       !item.item.itemCategoryHashes.includes(ItemCategoryHashes.RepeatableBounties));
+
+  const mod = item.item.itemCategoryHashes.includes(ItemCategoryHashes.Mods_Mod);
+
+  const unavailable = !item.canPurchase || !item.canBeSold || (owned && ownershipRule);
   return (
     <VendorItemDisplay
       item={item.item}
       unavailable={unavailable}
       owned={owned}
       acquired={acquired}
-      extraData={{ failureStrings: item.failureStrings, owned, acquired }}
```

Comment:
The `mod` part was from when mods were actual consumables and there was value in showing that a mod was present on some piece of gear. That code has bitrotted since then and `mod` was never set. The change isn't *great* either because it calls shaders "items" but ornaments and emotes "mods", so I'll file an issue to add accurate display names.

Reply:
Sounds great. I'd love to get rid of this "extraData" thing.